### PR TITLE
Config include trailing slash

### LINKED ISSUE #7951: Include configuration syntax for hidden folders
## Motivation

Hi, I've struggled (like a lot of persons it seems) with setting up github pages with shpinx. In particular, the way `include` works on folders/subfolders.

To solve the problem, I went to the documentation and looked how it was done in the default configuration [here](https://jekyllrb.com/docs/configuration/default/). And either the syntax is not the same for include or exclude, either there is a typo in the documentation:

To exclude a folder, it seems like one needs to write:
```
exclude: ["folder_name/"]
```
But it did not work to include a folder (specifically a folder starting with an underscore).
After some testing and looking around, I found that the syntax for  `include` seems to be:
```
include: ["_special_folder_name"]
```
 

## Suggestion

It is quite frustrating to have to guess that I find. Maybe an update of the documentation would be needed ?

A great way to improve the doc would be to give examples, at least for that case, but also maybe for and command/option (the default configuration is quite helpful for that, but it wasn't enough for me this time).

Thanks !

### PR DESCRIPTION
**Pull request by billykong**
This is a 🐛 bug fix.

- I've added tests
- The test suite passes locally

## Summary

`include` in the configuration file should include directory when trailing slash is present, e.g.  
```
# in _config.yml
include:
  - _pages/
```
should include the directory `_pages/` if exists.   
  
The current behavior in the `master` is that including `_pages` will include the directory while `_pages/` does not.  
  
This PR check that `_pages/` matches a directory named `_pages` but not file.

## Context

Resolves #7951

### CODE REVIEW

**Comment by billykong** [comment]
Hi @DirtyF, is there any problem with this pull request that needs to be fixed or something I should do? It is my first time going through the pull request workflow in OSS. Please feel free to let me know if I missed something.

**Comment by DirtyF** [comment]
@billykong nope, we're good, we just need to take the time to review your submission.

File: test/test_entry_filter.rb [hunk 1]
```diff
@@ -149,5 +149,12 @@ class TestEntryFilter < JekyllUnitTest
       assert @filter.glob_include?(data, "vendor/bundle/jekyll/lib/page.rb")
       assert @filter.glob_include?(data, "/vendor/ruby/lib/set.rb")
     end
+
+    should "match directory only if there is trailing slash" do
+      data = ["../_glob_include_test/_is_dir/", "../_glob_include_test/_not_dir/"]
```
Comment (ashmaroli) [inline test/test_entry_filter.rb:154]: our test site is in `test/source`. Please place the two `_glob_include_test` files in `test/source`.

File: test/test_entry_filter.rb [hunk 2]
```diff
@@ -149,5 +149,12 @@ class TestEntryFilter < JekyllUnitTest
...
+      assert @filter.glob_include?(data, "../_glob_include_test/_is_dir")
+      assert @filter.glob_include?(data, "../_glob_include_test/_is_dir/include_me.txt")
+      assert !@filter.glob_include?(data, "../_glob_include_test/_not_dir")
```
Comment (ashmaroli) [inline test/test_entry_filter.rb:157]: We prefer `refute something` instead of `assert !something`

**Comment by billykong** [comment]
Thanks @DirtyF and @ashmaroli

**Comment by ashmaroli** [comment]
@billykong Can you test separately if the issue you're resolving here can be reproduced with Jekyll 3.8.6..? (Want to see if this is a regression in Jekyll 4.0)..
Thanks.

**Comment by billykong** [comment]
> can be reproduced with Jekyll 3.8.6..?

Hi @ashmaroli , it can be reproduced in Jekyll 3.8.6

**Comment by billykong** [comment]
Tested with Jekyll 3.8.6.  
  
With Trailing slash directory:  
jekyll-3 8 6-with-trailing-slash-marked  
The `_pages` directory is not generated.
  
Without Trailing slash directory:  
jekyll-3 8 6-without-trailing-slash-marked
The `_pages` directory is generated.

**Comment by ashmaroli** [comment]
Thanks for testing with Jekyll 3.8.6 @billykong, So it is certain now that the bug existed prior to Jekyll 4.0 and Jekyll 3.8.6 :+1:

**Comment by ashmaroli** [comment]
Thanks @billykong 
@jekyllbot: merge +bug

**Pull request merged**